govcd/api_vcd_versions.go

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"sort"
+
+	semver "github.com/hashicorp/go-version"
+
+	"github.com/vmware/go-vcloud-director/v2/util"
+)
+
+type VersionInfo struct {
+	Version    string `xml:"Version"`
+	LoginUrl   string `xml:"LoginUrl"`
+	Deprecated bool   `xml:"deprecated,attr,omitempty"`
+}
+
+type VersionInfos []VersionInfo
+
+type SupportedVersions struct {
+	VersionInfos `xml:"VersionInfo"`
+}
+
+// APIVCDMaxVersionIs compares against maximum vCD supported API version from /api/versions (not necessarily
+// the currently used one). This allows to check what is the maximum API version that vCD instance
+// supports and can be used to guess vCD product version. API 31.0 support was first introduced in
+// vCD 9.5 (as per https://code.vmware.com/doc/preview?id=8072). Therefore APIMaxVerIs(">= 31.0")
+// implies that you have vCD 9.5 or newer running inside.
+// It does not require for the client to be authenticated.
+//
+// Format: ">= 27.0, < 32.0", ">= 30.0", "= 27.0"
+//
+// vCD version mapping to API version support https://code.vmware.com/doc/preview?id=8072
+func (vdcCli *VCDClient) APIVCDMaxVersionIs(versionConstraint string) bool {
+	err := vdcCli.vcdFetchSupportedVersions()
+	if err != nil {
+		util.Logger.Printf("[ERROR] could not retrieve supported versions: %s", err)
+		return false
+	}
+
+	util.Logger.Printf("[TRACE] checking max API version against constraints '%s'", versionConstraint)
+	maxVersion, err := vdcCli.maxSupportedVersion()
+	if err != nil {
+		util.Logger.Printf("[ERROR] unable to find max supported version : %s", err)
+		return false
+	}
+
+	isSupported, err := vdcCli.apiVerMatchesConstraint(maxVersion, versionConstraint)
```

Comment:
I would like to use full words `Ver` -> `Version`

Reply:
Done